tests/shipment_test.py

```diff
@@ -0,0 +1,154 @@
+import pytest
+
+import easypost
+import tests.fixture as fixture
+
+
+@pytest.mark.vcr()
+def test_shipment_create():
+    shipment = easypost.Shipment.create(**fixture.full_shipment())
+
+    assert isinstance(shipment, easypost.Shipment)
+    assert str.startswith(shipment.id, "shp_")
+    assert shipment.rates is not None
+    assert shipment.options.label_format == "PNG"
+    assert shipment.options.invoice_number == "123"
+    assert shipment.reference == "123"
+
+
+@pytest.mark.vcr()
+def test_shipment_retrieve():
+    shipment = easypost.Shipment.create(**fixture.full_shipment())
+
+    retrieved_shipment = easypost.Shipment.retrieve(shipment.id)
+
+    assert isinstance(retrieved_shipment, easypost.Shipment)
+    assert retrieved_shipment.id == shipment.id
+
+
+@pytest.mark.vcr()
+def test_shipment_all():
+    shipments = easypost.Shipment.all(page_size=fixture.page_size())
+
+    shipment_array = shipments["shipments"]
+
+    assert len(shipment_array) <= fixture.page_size()
+    assert shipments["has_more"] is not None
+    assert all(isinstance(shipment, easypost.Shipment) for shipment in shipment_array)
+
+
+@pytest.mark.vcr()
+def test_shipment_buy():
+    shipment = easypost.Shipment.create(**fixture.full_shipment())
+    shipment.buy(rate=shipment.lowest_rate())
+
+    assert shipment.postage_label is not None
+
+
+@pytest.mark.vcr()
+def test_shipment_regenerate_rates():
+    shipment = easypost.Shipment.create(**fixture.full_shipment())
+
+    rates = shipment.regenerate_rates()
+
+    rates_array = rates["rates"]
+
+    assert isinstance(rates_array, list)
+    assert all(isinstance(rate, easypost.Rate) for rate in rates_array)
+
+
+@pytest.mark.vcr()
+def test_shipment_convert_label():
+    shipment = easypost.Shipment.create(**fixture.full_shipment())
+    shipment.buy(rate=shipment.lowest_rate())
+    shipment.label(file_format="ZPL")
+
+    assert shipment.postage_label.label_zpl_url is not None
+
+
+# If the shipment was purchased with a USPS rate, it must have had its insurance set to `0` when bought
```

Comment:
In Python, these large block comments should instead be docblocks, please move them just below the test function name and wrap them in trouble quotes. This is true for all the test comments that are at the "function" level.